core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -444,7 +444,9 @@ class CollectionAccumulator[T] extends AccumulatorV2[T, java.util.List[T]] {
 
   override def copy(): CollectionAccumulator[T] = {
     val newAcc = new CollectionAccumulator[T]
-    newAcc._list.addAll(_list)
+    _list.synchronized {
```

Comment:
(context) synchronizedList is not protected by 'this', but by a private variable - so the locks are different.